openpype/pipeline/workfile/workfile_template_builder.py

```diff
@@ -1246,6 +1247,31 @@ def get_load_plugin_options(self, options=None):
 
         loader_items = list(sorted(loader_items, key=lambda i: i["label"]))
         options = options or {}
+
+        # Get families from all loaders excluding "*"
+        all_loaders = discover_loader_plugins()
+        families = set()
+        representations = set()
+        for loader in all_loaders:
+            families.update(loader.families)
+        families.discard("*")
+
+        # Sort for readability
+        families = list(sorted(families))
+
+        # Get representations for the default family,
+        # or the first one if no default
+        attributes = ['representations', 'extensions', 'extension']
+        for loader in all_loaders:
+            for family in loader.families:
+                if family == (options.get('family') or families[0]):
+                    for attribute in attributes:
+                        if hasattr(loader, attribute):
+                            representations.update(getattr(loader, attribute))
+
+        representations.discard("*")
+        representations = list(sorted(representations))
```

Comment:
I would personally move `representations = set()` definition down closer to where it's used. But not sure if that's just my personal preference?

```suggestion
        for loader in all_loaders:
            families.update(loader.families)
        families.discard("*")

        # Sort for readability
        families = list(sorted(families))

        # Get representations for the default family,
        # or the first one if no default
        attributes = ['representations', 'extensions', 'extension']
        representations = set()
        for loader in all_loaders:
            for family in loader.families:
                if family == (options.get('family') or families[0]):
                    for attribute in attributes:
                        if hasattr(loader, attribute):
                            representations.update(getattr(loader, attribute))

        representations.discard("*")
        representations = list(sorted(representations))
```

Also, since we're re-using the logic. Shouldn't we move the `get_representation_by_family` logic into some sort of library function?

Reply:
I agree with converting this method into a library function. But where could it fit in ?